File: platforms/ios/Sources/WhiskerModule/WhiskerModule.swift
/// Marks a concrete `Module` declaration for Whisker's generated registry.
///
/// The macro itself emits no runtime code. The SwiftPM build-tool plugin finds
/// the marker and generates one registration function for the containing
/// target, matching Rust's and Android's explicit declaration model.
@attached(peer)
public macro WhiskerModule() =
    #externalMacro(module: "WhiskerModuleMacros", type: "WhiskerModuleMacro")

File: platforms/android/module/src/main/kotlin/rs/whisker/runtime/WhiskerModule.kt
package rs.whisker.runtime

/**
 * Marks a concrete [Module] declaration for Whisker's generated registry.
 *
 * The annotation is the explicit registration signal shared with the Rust and
 * Swift module APIs. KSP validates that the annotated class extends [Module].
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
public annotation class WhiskerModule
